### PR TITLE
Add support for uLanding radar altimeter

### PR DESCRIPTION
**Pull request by davidaroyer**
These changes to support Aerotenna's [uLanding](http://aerotenna.com/sensors/#ulanding) radar altimeter have been tested on both a pixhawk and [Aerotenna's OcPoC-Altera](http://aerotenna.com/ocpoc-cyclone/).

### CODE REVIEW

**Comment by lvale** [comment]
@davidaroyer Any more info regarding also the uSharp sensor ? I might be interested on building a solution for that.

File: libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    void update(void);
+
+private:
+    // get a reading
+    bool get_reading(uint16_t &reading_cm);
+
+    AP_HAL::UARTDriver *uart = nullptr;
+    uint32_t last_reading_ms = 0;
+    uint8_t linebuf[10];
+    uint8_t linebuf_len = 0;
```
Comment (magicrub) [inline libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h]: In C+11 you don't need to initialize to zero, including =nullptr.

File: libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// -*- tab-width: 4; Mode: C++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+#pragma once
+
+#include "RangeFinder.h"
+#include "RangeFinder_Backend.h"
+
+#define ULANDING_COEFF 2.5
```
Comment (magicrub) [inline libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h]: Is this actually supposed to be 2.54?

Comment (magicrub) [inline libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h]: Also, needs an f on the end like 2.5f

File: libraries/AP_SerialManager/AP_SerialManager.h [hunk 3]
```diff
@@ -63,6 +63,12 @@
 #define AP_SERIALMANAGER_SToRM32_BUFSIZE_RX     128
 #define AP_SERIALMANAGER_SToRM32_BUFSIZE_TX     128
 
+// Aerotenna uLanding Altimeter
+// Note that size of UART FIFO is 128 for Altera-OcPoc board
+#define AP_SERIALMANAGER_ULANDING_BAUD           115200
+#define AP_SERIALMANAGER_ULANDING_BUFSIZE_RX     128
+#define AP_SERIALMANAGER_ULANDING_BUFSIZE_TX     128
```
Comment (magicrub) [inline libraries/AP_SerialManager/AP_SerialManager.h]: What does the 128 byte tx buffer do? Seems like nothing. Can this be safely changed to.. 1?

Comment (magicrub) [inline libraries/AP_SerialManager/AP_SerialManager.h]: Or 0?

File: libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h (hunk 1 above)
Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h]: in ArduPilot anyway (as we override new). In general C++ you would need to initialise to zero, but we avoid that to save flash space

File: libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h (hunk 2 above)
Comment (tridge) [inline libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h]: note that we now pass a compiler flag to make float constants default to single precision, so we don't need to be careful any more.

**Comment by rmackay9** [comment]
Merged along with some formatting fixes, one of which addresses the "f" issue from TomP.

**Pull request closed without merging**